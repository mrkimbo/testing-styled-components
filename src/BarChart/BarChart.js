import React, { PureComponent } from 'react';
import { BarChartWrapper } from './BarChart.styles';
import BarChartItem from '../BarChartItem';

// util
const getChannelCount = (items = []) =>
  items.reduce((sum, { totalCount = 0 }) => (sum + totalCount), 0);

class BarChart extends PureComponent {
  render() {
    const { data = [] } = this.props;
    
    // Reorder the object according to the id
    data.sort((a, b) => (a.id > b.id) ? 1 : -1);
    
    // Create percentage data
    const totalChannelCount = getChannelCount(data);
    const renderData = data.map((item) => ({
      ...item,
      percentage: (item.totalCount * 100 / totalChannelCount)
    }));
    
    // Render chart items
    const chartItems = renderData.map((item) => (
      <BarChartItem key={item.name} {...item} />
    ));
    
    return (
      <BarChartWrapper>
        {chartItems}
      </BarChartWrapper>
    );
  }
}

BarChart.displayName = 'BarChart';
BarChart.propTypes = {};
BarChart.defaultProps = {
  data: [],
};

export default BarChart;
