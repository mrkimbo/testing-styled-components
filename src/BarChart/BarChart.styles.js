import styled from 'styled-components';

export const BarChartWrapper = styled.div`
  width: 100%;
  max-width: 190px;
`;
