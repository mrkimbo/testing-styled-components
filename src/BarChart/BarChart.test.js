import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import { BarChartWrapper } from './BarChart.styles';

import BarChart from "./BarChart";

const generateData = (item, idx, arr) => ({
  name: `item${idx}`,
  totalCount: Math.abs(arr.length - idx),
});

describe('<BarChart/>', () => {
  const dummyData = new Array(2).fill(null).map(generateData);
  const createComponent = (serializer = shallow) => serializer(
    <BarChart data={dummyData}/>
  );
  
  it('should match shallow snapshot', () => {
    expect(createComponent()).toMatchSnapshot();
  });
});

describe('[Styles]', () => {
  it('<BarChartWrapper/> should match snapshot', () => {
    expect(shallow(<BarChartWrapper/>)).toMatchSnapshot();
  });
});
