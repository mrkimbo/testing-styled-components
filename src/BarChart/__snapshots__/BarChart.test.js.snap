// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarChart/> should match shallow snapshot 1`] = `
<styled.div>
  <BarChartItem
    key="item0"
    name="item0"
    percentage={66.66666666666667}
    totalCount={2}
  />
  <BarChartItem
    key="item1"
    name="item1"
    percentage={33.333333333333336}
    totalCount={1}
  />
</styled.div>
`;

exports[`[Styles] <BarChartWrapper/> should match snapshot 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  width: 100%;
  max-width: 190px;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
